osquery/tables/system/windows/windows_security_products.cpp

```diff
@@ -0,0 +1,220 @@
+#include <codecvt>
+#include <locale>
+
+#include <iwscapi.h>
+#include <wscapi.h>
+
+#include <osquery/core.h>
+#include <osquery/core/windows/wmi.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+typedef struct wsc_entry {
+  WSC_SECURITY_PROVIDER provider;
+  std::wstring product_name;
+  WSC_SECURITY_PRODUCT_STATE product_state;
+  std::wstring product_state_timestamp;
+  std::wstring remediation_path;
+  WSC_SECURITY_SIGNATURE_STATUS signature_status;
+} wsc_entry;
+
+HRESULT GetSecurityProducts(WSC_SECURITY_PROVIDER provider,
+                            std::list<wsc_entry>& out_list) {
```

Comment:
Two things.
1. What do you use `std::list` ? It's the worst stl structure in terms of performance as far as I know :) Would it be better to use just `std::vector` here?
2. Please for osquery functions and methods use `Expected` to return `value-or-error`.

Reply:
fwiw I don't think your comment about the performance of `std::list` applies in this case...

From [cppreference](https://en.cppreference.com/w/cpp/container/list):

> std::list is a container that supports constant time insertion and removal of elements from anywhere in the container. Fast random access is not supported.

We're not using random access so I think the perf will be fine (and likely more space efficient than a `std::vector`). Nevertheless I can easily change it.

Reply:
Additionally, if I use `std::vector` and `Expected`, we will end up with a linear time append (https://en.cppreference.com/w/cpp/container/vector/insert) of each list rather than constant time. I'm going to change to returning `Status` for now and we can discuss further if you'd like to work something else out.

Reply:
Agree with returning vector, this sounds reasonable.

About `Status` vs `Expected` - we gonna move everything to `Expected` so, I'd suggest to use it where it is possible instead of `Status`. But it is up to you.

Reply:
To be clear, you are suggesting you would prefer to use the less performant version?

Reply:
I think he meant returning Status, and has no comment on the use of list.